### PR TITLE
Bridge driver changes to support IPAM model

### PR DESCRIPTION
**Pull request by aboch**

### CODE REVIEW

File: drivers/bridge/bridge.go [hunk 1]
```diff
@@ -49,17 +56,17 @@ type configuration struct {
...
 	EnableIPv6         bool
 	EnableIPMasquerade bool
 	EnableICC          bool
 	Mtu                int
-	DefaultGatewayIPv4 net.IP
-	DefaultGatewayIPv6 net.IP
 	DefaultBindingIP   net.IP
 	DefaultBridge      bool
+	// Internal fields set after ipam data parsing
+	addressIPv4        *net.IPNet
```
Comment (mrjana) [inline drivers/bridge/bridge.go]: Can you make this exported so that it helps in bridge driver json marshalling?

**Comment by mavenugo** [comment]
phase-2 changes LGTM.
(I implemented the phase-1 changes :) .... so will let others review it).

File: drivers/bridge/bridge.go [hunk 2]
```diff
@@ -516,6 +457,32 @@ func parseNetworkGenericOptions(data interface{}) (*networkConfiguration, error)
...
+	if ipamV4Data[0].Gateway != nil {
+		c.AddressIPv4 = types.GetIPNetCopy(ipamV4Data[0].Gateway)
+	}
+
+	gw, ok := ipamV4Data[0].AuxAddresses[DefaultGatewayV4AuxKey]
+	if ok {
+		c.DefaultGatewayIPv4 = gw.IP
+	}
+
+	gw, ok = ipamV4Data[0].AuxAddresses[DefaultGatewayV6AuxKey]
```
Comment (mavenugo) [inline drivers/bridge/bridge.go]: this should be ipamV6Data ?

File: drivers/bridge/bridge.go [hunk 3]
```diff
@@ -516,6 +457,32 @@ func parseNetworkGenericOptions(data interface{}) (*networkConfiguration, error)
...
+		return types.ForbiddenErrorf("bridge driver doesnt support multiple subnets")
+	}
+
+	if len(ipamV4Data) == 0 {
+		return types.BadRequestErrorf("bridge network %s requires ipv4 configuration", id)
+	}
+
+	if ipamV4Data[0].Gateway != nil {
+		c.AddressIPv4 = types.GetIPNetCopy(ipamV4Data[0].Gateway)
+	}
```
Comment (mavenugo) [inline drivers/bridge/bridge.go]: missing c.AddressIPv6 assignment

File: drivers/bridge/bridge.go (hunk 2 above)
Comment (aboch) [inline drivers/bridge/bridge.go]: Good catch.

File: drivers/bridge/bridge.go (hunk 3 above)
Comment (aboch) [inline drivers/bridge/bridge.go]: Thanks.

**Comment by mrjana** [comment]
LGTM

**Pull request merged**